scripts/jenkins/Jenkinsfile

```diff
@@ -1,5 +1,33 @@
 platforms = ["centos7", "debian9", "ubuntu16", "ubuntu18"]
 
+/*  The detectChanges function looks at the last git commit and searches for changes to
+*   "redhat/freeradius.spec" or anything in the "debian" directory.  If it 
+*   finds any changes to those files it returns the string " --no-cache " 
+*   which can be fed to a docker build command to tell docker to rebuild 
+*   the image from scratch so that it can pick up any new dependencies 
+*   specified in those package manifest files
+*/
+
+def detectChanges() {
+    def string = ""
+    def changeLogSets = currentBuild.changeSets
+    for (int i = 0; i < changeLogSets.size(); i++) {
+        def entries = changeLogSets[i].items
+        for (int j = 0; j < entries.length; j++) {
+            def entry = entries[j]
+            def files = new ArrayList(entry.affectedFiles)
+            for (int k = 0; k < files.size(); k++) {
+                def file = files[k]
+                if (file.path =~ /(^debian\/.*)|(redhat\/freeradius.spec)/) {
+                    echo "changes in file ${file.path}, passing --no-cache flag to docker build"
+                    string = " --no-cache "
```

Comment:
Due to being into two of `for (...)` levels. Probably should be better to just `return " --no-cache"` and get off from the `for()` instead of running through all entries even if ready matched.